### PR TITLE
fix(challenges): Call resetQuiz() when dashedName of the quiz changes

### LINKED ISSUE #16801: Multiple choice challenge bug
Steps to reproduce: 

1. Start a multiple choice question challenge and answer one or more of the questions.
2. On the map, choose another multiple choice question challenge

You'll notice that whichever question you were on in the previous challenge, you'll stay on the subsequent challenge. 

Here's a gif that demonstrates this:

multiple choice bug mov

### PR DESCRIPTION
**Pull request by YogeshSharma0201**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #16801

#### Description
'currentIndex', 'selectedChoice' and 'correct' state variables of the Quiz
component were not changing when navigating to other quiz challenges
through map. Fixed it by calling resetQuiz() whenever dashedName of
the quiz changes.

### CODE REVIEW

File: common/app/routes/Challenges/views/quiz/Quiz.jsx [hunk 1]
```diff
@@ -62,13 +62,13 @@ const propTypes = {
   correct: PropTypes.number,
   currentIndex: PropTypes.number,
   dashedName: PropTypes.string,
-  description: PropTypes.string,
+  description: PropTypes.array,
   meta: PropTypes.object,
-  nextQuestion: PropTypes.fun,
-  resetChoice: PropTypes.fun,
-  resetQuiz: PropTypes.fun,
+  nextQuestion: PropTypes.func,
```
Comment (Bouncey) [inline common/app/routes/Challenges/views/quiz/Quiz.jsx]: nice catch :+1:

File: common/app/routes/Challenges/views/quiz/Quiz.jsx [hunk 2]
```diff
@@ -62,13 +62,13 @@ const propTypes = {
   correct: PropTypes.number,
   currentIndex: PropTypes.number,
   dashedName: PropTypes.string,
-  description: PropTypes.string,
+  description: PropTypes.array,
```
Comment (Bouncey) [inline common/app/routes/Challenges/views/quiz/Quiz.jsx]: This should really be `PropTypes.arrayOf(PropTypes.string)`

**Review by Bouncey — CHANGES_REQUESTED** [review]

File: common/app/routes/Challenges/views/quiz/Quiz.jsx (hunk 2 above)
Comment (YogeshSharma0201) [reply common/app/routes/Challenges/views/quiz/Quiz.jsx]: Hi @Bouncey ! 
The description object is actually is an array of objects and not strings.
screenshot from 2018-03-07 17-51-39

So, should i change it to PropTypes.arrayOf(PropTypes.object) ? or something else?

Comment (Bouncey) [reply common/app/routes/Challenges/views/quiz/Quiz.jsx]:
```js
const propTypes = {
  // ...
  description: PropTypes.arrayOf(
    PropTypes.shape({
      answer: PropTypes.number,
      choices: PropTypes.arrayOf(PropTypes.string),
      explanation: PropTypes.string
      question: PropTypes.string
      subtitle: PropTypes.string
    })
  ),
  // ...
};
```

**Comment by YogeshSharma0201** [comment]
@Bouncey i have made the changes requested.

**Comment by Bouncey** [comment]
Great work! Thanks @YogeshSharma0201 :+1:

**Review by Bouncey — APPROVED** [review]

**Pull request merged**